src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyResultSet.java

```diff
@@ -625,4 +629,175 @@ public void storeAssemblyHaplotypes() {
     public boolean hasOverwrittenHaps() {
         return !originalAssemblyHaps.isEmpty();
     }
+
+    /**
+     * Remove haplotypes with alleles we wish to filter
+     * // TODO this is a bad algorithm for bad people -- it might eliminate good alleles on the same haplotypes
+     * // TODO: this should be a method of AssemblyResultSet, not an external static method
+     */
+    public void removeHaplotypesWithBadAlleles(final AssemblyBasedCallerArgumentCollection argumentCollection,
+                                               final List<BiallelicVariant> badPileupEvents) {
+        if (badPileupEvents.isEmpty()) {
+            return; // nothing to do
+        }
+        List<Haplotype> haplotypesWithFilterAlleles = new ArrayList<>();
+        // If we are not using partially determined haplotypes, we discard every haplotype containing a bad pileup allele.
+
+
+        for(BiallelicVariant badEvent : badPileupEvents) {
+            for (Haplotype hap : getHaplotypeList()) {
+                // NOTE: The event map may be null due to edge cases in the assembly engine + SW realignment that can cause
+                //       haplotypes to re-merge into the reference and end up with an empty event map. (Also the event map
+                //       code explicitly expects to fail in some instances)
+                if (hap.getEventMap() != null && hap.getEventMap().getEvents().stream().anyMatch(badEvent::sameRefAltAndStart)) {
+                    if (argumentCollection.pileupDetectionArgs.debugPileupStdout) System.err.println("Flagging hap " + hap + " for containing variant " + badEvent);
+                    haplotypesWithFilterAlleles.add(hap);
+                }
+            }
+        }
+
+        if (!haplotypesWithFilterAlleles.isEmpty()) {
+            if (argumentCollection.pileupDetectionArgs.debugPileupStdout) System.out.println("Found Assembly Haps with filtered Variants:\n"+haplotypesWithFilterAlleles.stream().map(Haplotype::toString).collect(Collectors.joining("\n")));
+            AssemblyBasedCallerUtils.haplotypeDeletionWarningLogger.warn(() -> "Haplotypes from Assembly are being filtered by heuristics from the PileupCaller. This can lead to missing variants. See --"+PileupDetectionArgumentCollection.PILEUP_DETECTION_FILTER_ASSEMBLY_HAPS_THRESHOLD+" for more details");
```

Comment:
You might as well move the warning logger to this while you are moving things.

Reply:
done